roles/haproxy/files/haproxy.tmpl

```diff
@@ -1,15 +1,25 @@
 global
-    maxconn 256
+    maxconn {{or (key "service/haproxy/maxconn") 256}}
     debug
 
 defaults
     mode http
-    timeout connect 5000ms
-    timeout client 50000ms
-    timeout server 50000ms
+    timeout connect {{or (key "service/haproxy/timeouts/connect") "5000ms"}}
+    timeout client {{or (key "service/haproxy/timeouts/client") "50000ms"}}
+    timeout server {{or (key "service/haproxy/timeouts/server") "50000ms"}}
+
+# frontend used to return health status without requiring SSL
+frontend haproxy_status
+    bind 0.0.0.0:34180      # 34180 means EALTH ;)
+    # create a status URI in /haproxy_status that will return
+    # a 200 is backend is healthy, and 503 if it isn't. This
+    # URI is queried by the ELB.
+    acl backend_dead nbsrv(consul_backend) lt 1
+    monitor-uri /haproxy_status
+    monitor fail if backend_dead
 
 frontend www
-    bind *:80
+    bind *:80 accept-proxy
```

Comment:
There appears to be a problem with this line here (when running up in the vagrant machine/aws, so removing it for now)